manifests/charts/istio-control/istio-discovery/files/kube-gateway.yaml

```diff
@@ -3,16 +3,20 @@ kind: ServiceAccount
 metadata:
   name: {{.ServiceAccount | quote}}
   namespace: {{.Namespace | quote}}
+  annotations:
+    {{- toJsonMap (omit .InfrastructureAnnotations "kubectl.kubernetes.io/last-applied-configuration" "gateway.istio.io/name-override" "gateway.istio.io/service-account" "gateway.istio.io/controller-version") | nindent 4 }}
+  labels:
+    {{- toJsonMap .InfrastructureLabels | nindent 4 }}
 ---
 apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: {{.DeploymentName | quote}}
   namespace: {{.Namespace | quote}}
   annotations:
-    {{- toJsonMap (omit .Annotations "kubectl.kubernetes.io/last-applied-configuration" "gateway.istio.io/name-override" "gateway.istio.io/service-account" "gateway.istio.io/controller-version") | nindent 4 }}
+    {{- toJsonMap (omit .InfrastructureAnnotations "kubectl.kubernetes.io/last-applied-configuration" "gateway.istio.io/name-override" "gateway.istio.io/service-account" "gateway.istio.io/controller-version") | nindent 4 }}
   labels:
-    {{- toJsonMap .Labels | nindent 4 }}
+    {{- toJsonMap .InfrastructureLabels | nindent 4 }}
```

Comment:
I think we probably want to merge with .Labels for one release for backwards compat

Reply:
nvm, I see the logic in the code and prefer it to what I recommended